ouroboros-consensus/src/Ouroboros/Consensus/Storage/LedgerDB/HD/DiffSeq.hs

```diff
@@ -0,0 +1,281 @@
+{-# LANGUAGE ConstraintKinds            #-}
+{-# LANGUAGE DataKinds                  #-}
+{-# LANGUAGE DeriveAnyClass             #-}
+{-# LANGUAGE DeriveFunctor              #-}
+{-# LANGUAGE DeriveGeneric              #-}
+{-# LANGUAGE DerivingVia                #-}
+{-# LANGUAGE FlexibleContexts           #-}
+{-# LANGUAGE FlexibleInstances          #-}
+{-# LANGUAGE GeneralisedNewtypeDeriving #-}
+{-# LANGUAGE MultiParamTypeClasses      #-}
+{-# LANGUAGE ScopedTypeVariables        #-}
+{-# LANGUAGE TypeFamilies               #-}
+
+{-# LANGUAGE UndecidableInstances       #-}
+
+module Ouroboros.Consensus.Storage.LedgerDB.HD.DiffSeq (
+    -- * Sequences of diffs
+    DiffSeq (..)
+  , Element (..)
+  , InternalMeasure (..)
+  , Length (..)
+  , RootMeasure (..)
+  , SlotNo (..)
+    -- * Short-hands for type-class constraints
+  , SM
+    -- * API: derived functions
+  , cumulativeDiff
+  , extend
+  , extend'
+  , length
+  , mapDiffSeq
+  , maxSlot
+  , splitl
+  , splitlAt
+  , splitlAtFromEnd
+  , splitr
+  , splitrAt
+  , splitrAtFromEnd
+  ) where
+
+import           Prelude hiding (length, splitAt)
+
+import qualified Control.Exception as Exn
+import           Data.Bifunctor (Bifunctor (bimap))
+import           Data.Group
+import           Data.Monoid (Sum (..))
+import           Data.Semigroup (Max (..))
+import           GHC.Generics (Generic)
+import           NoThunks.Class (NoThunks)
+
+import           Data.FingerTree.RootMeasured.Strict hiding (split, splitl,
+                     splitlr, splitr)
+import qualified Data.FingerTree.RootMeasured.Strict as RMFT (splitlr)
```

Comment:
Do we need to explicitly import `splitlr` if we're using a qualified import?

Reply:
It's not necessary, but all other definitions are imported unqualified and the qualified import is really only necessary for `splitlr` because this module also defines a `splitlr` function